orangecontrib/text/widgets/owsentimentanalysis.py

```diff
@@ -0,0 +1,69 @@
+from PyQt5.QtWidgets import QApplication
+
+from Orange.widgets import gui
+from Orange.widgets.settings import Setting
+from Orange.widgets.utils.signals import Input, Output
+from Orange.widgets.widget import OWWidget
+from orangecontrib.text import Corpus
+from orangecontrib.text.sentiment import Vader_Sentiment
+
+
+class OWSentimentAnalysis(OWWidget):
+    name = "Sentiment Analysis"
+    description = "Predict sentiment from text."
+    icon = "icons/SentimentAnalysis"
+    priority = 30001
+
+    class Inputs:
+        corpus = Input("Corpus", Corpus)
+
+    class Outputs:
+        corpus = Output("Corpus", Corpus)
+
+    autocommit = Setting(True)
+    want_main_area = False
+
+    def __init__(self):
+        super().__init__()
+        self.corpus = None
+
+        info_box = gui.widgetBox(self.controlArea)
+        self.info_label = gui.label(info_box, self, 'n/a')
+        gui.rubber(self.controlArea)
+
+        gui.auto_commit(self.controlArea, self, 'autocommit', 'Commit',
+                        'Autocommit is on')
+
+    @Inputs.corpus
+    def set_corpus(self, data=None):
+        self.corpus = data
+        if data is not None and not isinstance(data, Corpus):
+            self.corpus = Corpus.from_table(data.domain, data)
```

Comment:
I wouldn't take care of this anymore. IMHO we should expect Corpus on input and not care about when a user provides a Table. Otherwise, this should be a part of all our widgets and I'm not sure if it really makes sense; e.g. you don't even know what features are used as text etc.